src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPInterfaceStatistics.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace System.Net.NetworkInformation
+{
+    /// <summary>
+    /// IPInterfaceStatistics provider for Linux. 
+    /// Reads information out of /proc/net/dev and other locations.
+    /// </summary>
+    internal class LinuxIPInterfaceStatistics : IPInterfaceStatistics
+    {
+        // /proc/net/dev
+        // Receive section
+        private uint _bytesReceived;
+        private uint _packetsReceived;
+        private uint _errorsReceived;
+        private uint _incomingPacketsDropped;
+        private uint _fifoBufferErrorsReceived;
+        private uint _packetFramingErrorsReceived;
+        private uint _compressedPacketsReceived;
+        private uint _multicastFramesReceived;
+
+        // Transmit section
+        private uint _bytesTransmitted;
+        private uint _packetsTransmitted;
+        private uint _errorsTransmitted;
+        private uint _outgoingPacketsDropped;
+        private uint _fifoBufferErrorsTransmitted;
+        private uint _collisionsDetected;
+        private uint _carrierLosses;
+        private uint _compressedPacketsTransmitted;
+
+        // From /sys/class/net/<interface>/tx_queue_len
+        private int _transmitQueueLength;
+
+        public LinuxIPInterfaceStatistics(string name)
+        {
+            ParseTable(name);
+            string transmitQueueLengthFilePath = Path.Combine(LinuxNetworkFiles.SysClassNetFolder, name, "tx_queue_len");
+            _transmitQueueLength = int.Parse(File.ReadAllText(transmitQueueLengthFilePath));
+        }
+
+        private void ParseTable(string name)
+        {
+            /* NOTE: This same information can be obtained using rtnetlink:
+                 RTM_GETLINK:
+                   Get information about a specific network interface
+                   rta_type:IFLA_STATS contains all of the below information.
+                   https://github.com/torvalds/linux/blob/dd5cdb48edfd34401799056a9acf61078d773f90/include/uapi/linux/if_link.h#L41
+                   There does not appear to be any additional information exposed there that is not exposed in /proc/net/dev
+            */
+
+            using (var sr = File.OpenText(LinuxNetworkFiles.InterfaceListingFile))
```

Comment:
Nit: var allowed only if the type is mentioned on the same line (e.g. `var file = new File()`).